packages/vite/src/node/plugins/wasm.ts

```diff
@@ -7,11 +7,19 @@ const wasmHelperId = '/__vite-wasm-helper'
 const wasmHelper = async (opts = {}, url: string) => {
   let result
   if (url.startsWith('data:')) {
-    // @ts-ignore
-    const binaryString = atob(url.replace(/^data:.*?base64,/, ''))
-    const bytes = new Uint8Array(binaryString.length)
-    for (let i = 0; i < binaryString.length; i++) {
-      bytes[i] = binaryString.charCodeAt(i)
+    const urlContent = url.replace(/^data:.*?base64,/, '')
+    let bytes
+    if (typeof Buffer === 'function' && typeof Buffer.from === 'function') {
+      bytes = Buffer.from(urlContent, 'base64')
+    } else if (typeof atob === 'function') {
```

Comment:
Should we check `atob` first, then `Buffer`? And perhaps we can put a note here to remove the `Buffer` condition when we drop node 14

Reply:
Actually I just read the PR description 😅  ignore me